src/game_api/script/usertypes/level_lua.cpp

```diff
@@ -31,10 +35,151 @@ void PostRoomGenerationContext::set_num_extra_spawns(std::uint32_t extra_spawn_i
+}
+void PostRoomGenerationContext::change_short_tile_code(SHORT_TILE_CODE short_tile_code, ShortTileCodeDef short_tile_code_def)
+{
+    State::get().ptr_local()->level_gen->data->change_short_tile_code(short_tile_code, short_tile_code_def);
+}
+
+std::optional<SHORT_TILE_CODE> PreHandleRoomTilesContext::get_short_tile_code(uint32_t tx, uint32_t ty, LAYER layer) const
+{
+    if (tx >= 0 && tx < 10 && ty >= 0 && ty < 8)
+    {
+        layer = layer < 0 ? 0 : layer;
+        if (layer == LAYER_FRONT)
+        {
+            return get_room_data().front_layer[ty][tx];
+        }
+        else if (has_back_layer())
+        {
+            return get_room_data().back_layer.value()[ty][tx];
+        }
+    }
+    return std::nullopt;
+}
+bool PreHandleRoomTilesContext::set_short_tile_code(uint32_t tx, uint32_t ty, LAYER layer, SHORT_TILE_CODE short_tile_code)
+{
+    if (tx >= 0 && tx < 10 && ty >= 0 && ty < 8)
+    {
+        if (layer == LAYER_BOTH)
+        {
+            set_short_tile_code(tx, ty, 1, short_tile_code);
+            layer = 0;
+        }
+
+        layer = layer < 0 ? 0 : layer;
+        if (layer == LAYER_FRONT)
+        {
+            get_mutable_room_data().front_layer[ty][tx] = short_tile_code;
+            return true;
+        }
+        else if (has_back_layer())
+        {
+            get_mutable_room_data().back_layer.value()[ty][tx] = short_tile_code;
+            return true;
+        }
+    }
+    return false;
+}
+std::vector<std::tuple<uint32_t, uint32_t, LAYER>> PreHandleRoomTilesContext::find_all_short_tile_codes(LAYER layer, SHORT_TILE_CODE short_tile_code)
+{
+    if (layer == LAYER_BOTH)
+    {
+        std::vector<std::tuple<uint32_t, uint32_t, LAYER>> positions = find_all_short_tile_codes(LAYER_FRONT, short_tile_code);
+        std::vector<std::tuple<uint32_t, uint32_t, LAYER>> positions_back = find_all_short_tile_codes(LAYER_BACK, short_tile_code);
+        positions.insert(positions.end(), positions_back.begin(), positions_back.end());
+        return positions_back;
+    }
+    layer = layer < 0 ? 0 : layer;
+    if (layer == LAYER_FRONT || has_back_layer())
+    {
+        const LevelGenRoomData& data = get_room_data();
+        const SingleRoomData& mutable_room_data = layer == LAYER_FRONT ? data.front_layer : data.back_layer.value();
+
+        std::vector<std::tuple<uint32_t, uint32_t, LAYER>> positions;
+        for (uint32_t tx = 0; tx < 10; tx++)
+        {
+            for (uint32_t ty = 0; ty < 8; ty++)
+            {
+                if (mutable_room_data[ty][tx] == short_tile_code)
+                {
+                    positions.push_back({tx, ty, layer});
+                }
+            }
+        }
+        return positions;
+    }
+    return {};
+}
+bool PreHandleRoomTilesContext::replace_short_tile_code(LAYER layer, SHORT_TILE_CODE short_tile_code, SHORT_TILE_CODE replacement_short_tile_code)
+{
+    layer = layer < 0 ? 0 : layer;
```

Comment:
missing enum_to_layer, unless this can't be executed after players are spawned in?

Reply:
Happens during room spawning so naturally the player isn't spawned. I should add that to the docs if it's not already there.